crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,732 @@
+    // freezing that client.
+    for (counterparty_chain_id, counterparty_client_id) in counterparty_clients {
+        let misbehaviour = TendermintMisbehaviour {
+            client_id: counterparty_client_id.clone(),
+            header1: header1.clone(),
+            header2: header2.clone(),
+        };
+
+        if !chains.contains_key(chain.id()) {
+            let chain_handle = spawn_chain_runtime_with_modified_config::<BaseChainHandle>(
+                &config,
+                chain.id(),
+                rt.clone(),
+                |chain_config| {
+                    if let Some(key_name) = key_name {
+                        chain_config.key_name = key_name.to_string();
+                    }
+                },
+            )?;
+
+            chains.insert(chain.id().clone(), chain_handle);
+        }
+
+        if !chains.contains_key(&counterparty_chain_id) {
+            let chain_handle = spawn_chain_runtime_with_modified_config::<BaseChainHandle>(
+                &config,
+                &counterparty_chain_id,
+                rt.clone(),
+                |chain_config| {
+                    if let Some(key_name) = key_name {
+                        chain_config.key_name = key_name.to_string();
+                    }
+                },
+            )?;
+
+            chains.insert(counterparty_chain_id.clone(), chain_handle);
+        };
+
+        let chain_handle = chains.get(chain.id()).unwrap();
+        let counterparty_chain_handle = chains.get(&counterparty_chain_id).unwrap();
+
+        let counterparty_client = ForeignClient::restore(
+            counterparty_client_id.clone(),
+            counterparty_chain_handle.clone(),
+            chain_handle.clone(),
+        );
+
+        let result = submit_light_client_attack_evidence(
+            &evidence,
+            chain,
+            counterparty_client,
+            counterparty_client_id,
+            counterparty_chain_handle,
+            misbehaviour,
+        );
+
+        if let Err(error) = result {
+            error!("{error}");
+        }
+    }
+
+    Ok(())
+}
+
+fn submit_light_client_attack_evidence(
+    evidence: &LightClientAttackEvidence,
+    chain: &CosmosSdkChain,
+    counterparty_client: ForeignClient<BaseChainHandle, BaseChainHandle>,
+    counterparty_client_id: ClientId,
+    counterparty: &BaseChainHandle,
+    misbehaviour: TendermintMisbehaviour,
+) -> Result<(), eyre::Error> {
+    info!(
+        "building misbehaviour evidence for client `{}` on chain `{}`",
+        counterparty_client_id,
+        counterparty.id(),
+    );
+
+    let signer = counterparty.get_signer()?;
+    let common_height = Height::from_tm(evidence.common_height, chain.id());
+
+    if counterparty_client.is_frozen() {
+        warn!(
+            "skipping client `{}` on chain `{}` as it is already frozen",
+            counterparty_client_id,
+            counterparty.id()
+        );
+
+        return Ok(());
+    }
```

Comment:
We should remove from here, still need to submit `MsgSubmitIcsConsumerMisbehaviour` client is frozen. Make the check for `MsgSubmitMisbehaviour` instead.
Also, isn't this checked by the caller in `fetch_all_counterparty_clients`? But will need to remove from there anyway.